File: EQUALIZER/MATLAB/pruebas.m
% function pruebas
%%
clc;clear all;
Fs = 44100;
fc = 60;
fs = fc/(Fs/2);
fp = fc+3;
fp = fp/(Fs/2);
[n wn] = buttord( fp, fs, 0.5, 80 );
[b, a] = butter(5,wn);
%%


[ H, F ] = freqz( b, a, 512,  1 );
F = Fs * F;
H = abs(H);
plot( F, H );
axis( [0 Fs/256 0 (max(H)+1)] );

data = sprintf('\nconst int n = %i;\nconst double b[%i] = { ', length(b), length(b) );
for id = 1:length(b)
    if id==length(b)
        data = sprintf('%s%5.16f};', data, b(id) );
        break;
    else
        data = sprintf('%s%5.16f, ',  data, b(id) );
    end
end


data = sprintf('%s\nconst double a[%i] = {', data, length(a) );
for id = 1:length(a)
    if id==length(a)
        data = sprintf('%s%5.16f};\n\n\n', data, 0 );
        break;
    else
        data = sprintf('%s%5.16f, ',  data, a(id+1) );
    end
end
clipboard('copy', data)


return
%%
clc;clear all;
n = 2048;
Fs = 44100;

fc = 40;
fs = fc/(Fs/2);
fp = fc+5;
fp = fp/(Fs/2);
    
FcHz = fc;
Fc   = FcHz/(Fs/2);
Wn = Fc;
% b = fir1( n, Wn, 'stop' );
b = fir1( n, Wn, 'High' );

data = sprintf('\nconst int n = %i;\nconst double b[n] = { ', length(b) );
for id = 1:length(b)
    if id==length(b)
        data = sprintf('%s%5.32f};\n\n\n', data, b(id) );
        break;
    else
        data = sprintf('%s%5.32f, ',  data, b(id) );
    end
end
clipboard('copy', data)

%%
try %#ok<TRYNC>
    stop( player );
    delete(player)
end

clc;clear all;
Fs = 48000;
FsN = Fs/2;
Wp = 50/FsN;
Ws = 500/FsN;
[n, Wn] = buttord( Wp, Ws, 20, 60 );



[b,a] = butter( n, Wn );
clc;

fprintf('\nconst int n = %i;\nconst double b[n] = { ',...
                    max([length(b), length(a)]) );
for id = 1:max([length(b), length(a)])
    if id==length(b)
        fprintf('%5.32f};', b(id) );
        break;
    else
        fprintf('%5.32f, ', b(id) );
    end
end
fprintf('\nconst double a[n] = { ');
for id = 1:max([length(b), length(a)])
    if id==length(a)
        fprintf('%5.32f};\n\n\n\n\n\n', a(id) );
        break;
    else
        fprintf('%5.32f, ', a(id) );
    end
end

return
freqz( b, a, 512, 1000 ); 
title('n=5 Butterworth Lowpass Filter')


[y Fs] = wavread( 'X:\SegAll\CONOCIMIENTO\UNIVERSIDAD\TRATAMIENTO\PROYECTO\Abrete.wav' );
y = y( 1:length(y)/1);
Ts = 1/Fs;% wavplay( y, Fs );
Gz = tf( b, a, Ts, 'z' );
t = Ts * (0:(length( y )-1));
drawnow
tic;
y1 = filtfilt( b, a, y );
y1 = y1/max( abs(y1) );
% disp( 'Tiempo, ' )
% disp( toc )


% 
% %%AO
% player = audioplayer(y1, Fs);
% play(player);

File: EQUALIZER/PRUEBAS.cpp
BOOL PRUEBAS(void){
    SetWindowTextF( HW_Info, (char*)"PRUEBAS...");
    WCLS.PlaySong( (char*)"X:\\SegAll\\CONOCIMIENTO\\UNIVERSIDAD\\TRATAMIENTO\\PROYECTO\\Abrete.wav");
    //
    WCLS.ReSetVal();
    WCLS.NormBufY();
    double* y = WCLS.GetY();
    double* x = WCLS.GetX();
    int Size = WCLS.GetSize();


    const double bLPF[4] = { 0.0000000348544221, 0.0000001045632663, 0.0000001045632663, 0.0000000348544221};
    const double aLPF[4] = {-2.9869054238176194, 2.9738964412704219, -0.9869907386174257, 0.0000000000000000};

    /*
    const double bk[6] = { 0.9967200902731339, -4.9836004513656693, 9.9672009027313386, -9.9672009027313386, 4.9836004513656693, -0.9967200902731339};
    const double ak[6] = {-4.9934294008818014, 9.9737391847402161, -9.9606411051342398, 4.9737822596299432, -0.9934509383540842, 0.0000000000000000};
    //*/





const double bk[6] = { 0.9987228574842059, -4.9936142874210292, 9.9872285748420584, -9.9872285748420584, 4.9936142874210292, -0.9987228574842059};
const double ak[6] = {-4.9974440825868189, 9.9897795963996767, -9.9846742910990578, 4.9897861233476180, -0.9974473460614176, 0.0000000000000000};














    int n = 0;
    double* b = NULL;
    double* a = NULL;
    BOOL Kareoke = TRUE;
    IF Kareoke==TRUE THEN
        n = 6;
        b = (double*)&bk[0];
        a = (double*)&ak[0];
    ELSE
        n = 4;
        b = (double*)&bLPF[0];
        a = (double*)&aLPF[0];
    ENDIF;






double VR1 = 0;
double VR2 = 0;

SetWindowTextF( HW_Info, (char*)"Procesando...");
tic
    for(int i=n;i<Size;i++){
        VR1 = 0;
        VR2 = 0;
        for(int ix=0;ix<n;ix++){
            VR1 = VR1 + b[ix] * x[2*i-2*ix];
            VR2 = VR2 + a[ix] * y[2*i-2*ix-2];
        }
        y[2*i] = VR1 - VR2;
    }
    IF Kareoke==TRUE THEN
        for(int i=0;i<Size;i++){
            y[2*i+1] = y[2*i+1] - y[2*i];
            y[2*i] = y[2*i+1];
            //y[2*i+1] = y[2*i];
        }
    ELSE
        for(int i=0;i<Size;i++){
            y[2*i+1] = y[2*i];
        }
    ENDIF;
    //*/
    WCLS.NormBufY();
    WCLS.FillSBufY();
    //*/
    SetWindowTextF( HW_Info, (char*)"Ok\nToc:\n%f", toc);
    return TRUE;
}
